src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -73,6 +71,22 @@ public bool ContainsAnyDocumentOrProjectDiagnostics(ProjectId projectId)
                 }
 
                 return !projectState.IsEmpty();
+
+                bool ContainsAnyDocumentDiagnostics()
```

Comment:
consider making static, and just passing in the data you need.  that helps ensure no allocs.

Reply:
Ah, got used to C# 7